Assets/MRTK/Core/Services/MixedRealityToolkit.cs

```diff
@@ -1044,12 +1049,16 @@ private bool ExecuteOnAllServicesReverseOrder(Action<IMixedRealityService> execu
                     return false;
                 }
 
-                var services = MixedRealityServiceRegistry.GetAllServices();
-                int length = services.Count;
-
-                for (int i = length - 1; i >= 0; i--)
+                try
+                {
+                    foreach (var service in MixedRealityServiceRegistry.GetAllServices().Reverse())
```

Comment:
@keveleigh  and @wiwei  double checking -- what's our policy on foreach loops? Not sure where we landed, want to make sure there is no perf impact.

Reply:
it is easy enough to keep the for loop and catch a different exception 
whichever makes folks happy :)

Reply:
Yeah, I'm not against `foreach` but I think I'd prefer to keep the loop as it was originally.

Reply:
Sounds good. I will restore the loops and update the exception handling,

Reply:
foreach is fine - @luis-valverde-ms did some work a while back to check on this and I believe this hasn't been an issue since the non-year-named Unity releases.

Reply:
Okay posting back because I think my comment here is a bit confusing:

When I was responding to the foreach comment, I thought I was responding specifically to the question of “are foreachs safe to use compared to for”.

Absent any other context, this is true – the memory alloc associated with foreach I believe was fixed many Unity versions ago.

Looking at my comments there I was off on one of the areas, but on the Reverse() specifically, I believe this does alloc new memory (either for the list itself, or for an enumerator that will be able to enumerate the read-only collection but in reverse). So in this case in order to use foreach, you have to do some memory alloc-y stuff which isn’t great in an inner loop.

Reply:
For my understanding and future reference: for and foreach are comparable but a foreach reverse is worse performance wise than a for reverse?

Reply:
The foreach itself isn't the problem -  it is the .Reverse() call (which AFAIK allocs more memory to maintain enumeration state). I could be wrong on this, but this is true for the LINQ usage (which this isn't) and I assume that this would still be returning a new collection because the underlying one (GetAllServices()) returns a read only data structure.

Reply:
for loops restored.